### PR TITLE
cleanup and renaming

### PR DESCRIPTION
**Pull request by zyndiecate**
Currently it is a lot of movement within other repos. I will create PR's for those that need user-config soon. 

Here we go
- https://github.com/giantswarm/cli/pull/497
- https://github.com/giantswarm/client-go/pull/92
- https://github.com/giantswarm/app-def-transf-go/pull/20
- https://github.com/giantswarm/api/pull/199
- https://github.com/giantswarm/app-client-go/pull/63
- https://github.com/giantswarm/app-service/pull/634

### CODE REVIEW

**Comment by zyndiecate** [comment]
RFR including deps @ewoutp

File: service_definition.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
+package userconfig
```
Comment (ewoutp) [inline service_definition.go]: I assume there are no other changes in here than renaming & copying from other files.
Right?

File: service_name.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package userconfig
+
+import (
+	"encoding/json"
+	"regexp"
+	"strings"
+)
+
+var (
+	appNameRegExp = regexp.MustCompile("^[a-zA-Z0-9]{1}[a-z0-9A-Z_.-]{0,99}$")
```
Comment (ewoutp) [inline service_name.go]: `serviceNameRegExp`

File: service_name.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+type ServiceName string
+
+// ParseServiceName returns the name of the given definition if it exists. It
+// is does not exist, it generates an app name.
```
Comment (ewoutp) [inline service_name.go]: it generates a service name

File: service_name.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+// Equals returns true if the given ServiceName is equal to the other
+// given application name, false otherwise.
+func (an ServiceName) Equals(other ServiceName) bool {
+	return an == other
+}
+
+// Validate checks that the given ServiceName is a valid ServiceName.
+func (an ServiceName) Validate() error {
+	if an.Empty() {
+		return maskf(InvalidServiceNameError, "app name must not be empty")
```
Comment (ewoutp) [inline service_name.go]: service name ...

File: service_name.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	anStr := an.String()
+	if !appNameRegExp.MatchString(anStr) {
+		return maskf(InvalidServiceNameError, "app name '%s' must match regexp: %s", anStr, appNameRegExp)
```
Comment (ewoutp) [inline service_name.go]: service name ...

File: service_name_test.go [hunk 6]
```diff
@@ -6,7 +6,7 @@ import (
 	"github.com/giantswarm/user-config"
 )
 
-func TestValidAppNames(t *testing.T) {
+func TestValidServiceNames(t *testing.T) {
 	list := map[string]string{
 		"a":           "app name should be allowed to be normal single character",
```
Comment (ewoutp) [inline service_name_test.go]: Probably best to rename the descriptions also: "service name should..."

File: service_name_test.go [hunk 7]
```diff
@@ -18,7 +18,7 @@ func TestValidAppNames(t *testing.T) {
 	}
 
 	for name, reason := range list {
-		appName := userconfig.AppName(name)
+		appName := userconfig.ServiceName(name)
```
Comment (ewoutp) [inline service_name_test.go]: appName -> serviceName

File: service_name_test.go [hunk 8]
```diff
@@ -27,7 +27,7 @@ func TestValidAppNames(t *testing.T) {
 	}
 }
 
-func TestInvalidAppNames(t *testing.T) {
+func TestInvalidServiceNames(t *testing.T) {
 	list := map[string]string{
 		"":      "app name must not be empty",
```
Comment (ewoutp) [inline service_name_test.go]: "app name...

File: service_name_test.go [hunk 9]
```diff
@@ -44,19 +44,19 @@ func TestInvalidAppNames(t *testing.T) {
 	}
 
 	for name, reason := range list {
-		appName := userconfig.AppName(name)
+		appName := userconfig.ServiceName(name)
```
Comment (ewoutp) [inline service_name_test.go]: appName -> serviceName

File: v2_user_config_env_test.go [hunk 10]
```diff
@@ -83,9 +83,9 @@ func TestUnmarshalV2EnvStructEmpty(t *testing.T) {
 	}
 }
 
-func TestUnmarshalV2EnvFullApp(t *testing.T) {
+func TestUnmarshalV2EnvFullService(t *testing.T) {
 	// Test the validator for full apps containing both array and structs
```
Comment (ewoutp) [inline v2_user_config_env_test.go]: full services

File: v2_user_config_env_test.go [hunk 11]
```diff
@@ -112,9 +112,9 @@ func TestUnmarshalV2EnvFullApp(t *testing.T) {
 	}
 }
 
-func TestUnmarshalV2EnvFullAppUpperCase(t *testing.T) {
+func TestUnmarshalV2EnvFullServiceUpperCase(t *testing.T) {
 	// Test the validator for full apps containing both array and structs with uppercase env keys
```
Comment (ewoutp) [inline v2_user_config_env_test.go]: full services

**Comment by ewoutp** [comment]
I Like it!

Please fix the mentioned issues, then LGTM (this PR, deps separate)

File: service_definition.go (hunk 1 above)
Comment (zyndiecate) [inline service_definition.go]: I think I also moved the `ParseServicedefinition` method into here.

File: service_name.go (hunk 2 above)
Comment (zyndiecate) [inline service_name.go]: Done

File: service_name.go (hunk 3 above)
Comment (zyndiecate) [inline service_name.go]: Done

File: service_name.go (hunk 4 above)
Comment (zyndiecate) [inline service_name.go]: Done

File: service_name.go (hunk 5 above)
Comment (zyndiecate) [inline service_name.go]: Done

File: service_name_test.go (hunk 6 above)
Comment (zyndiecate) [inline service_name_test.go]: Done

File: service_name_test.go (hunk 7 above)
Comment (zyndiecate) [inline service_name_test.go]: Done

File: service_name_test.go (hunk 8 above)
Comment (zyndiecate) [inline service_name_test.go]: Done

File: service_name_test.go (hunk 9 above)
Comment (zyndiecate) [inline service_name_test.go]: Done

File: v2_user_config_env_test.go (hunk 10 above)
Comment (zyndiecate) [inline v2_user_config_env_test.go]: Done

File: v2_user_config_env_test.go (hunk 11 above)
Comment (zyndiecate) [inline v2_user_config_env_test.go]: Done

**Pull request merged**